omniduct/filesystems/webhdfs.py

```diff
@@ -133,11 +133,18 @@ def _dir(self, path):
                 replication=f['replication']
             )
 
-    def _mkdir(self, path, recursive):
-        raise NotImplementedError
+    def _mkdir(self, path, recursive, exist_ok):
+        if not path.endswith('/'):
```

Comment:
This check (for a slash on the end of the path) is only required for S3 due to the way that the S3 API works. Here, you can just use `path`. Also, you should respect the `recursive` and `exist_ok` flags.

The body of this function should be something like:
```
if not recursive and not self.isdir(self.path_basename(path)):
    raise IOError("No parent directory found for {}.".format(path))
if not exist_ok and self.exists(path):
    raise IOError("Path already exists at {}.".format(path))
self.__webhdfs.make_dir(path)
```

Note: The webhdfs API `make_dir` method is equivalent to `mkdir -p` where pre-existence is not considered to be an issue.

Reply:
made the changes